### ISSUE TITLE
Lab 9: Peer-review

### ISSUE THREAD

**Issue by loregrc** [giacomofantino/computational-intelligence#4]
Hi Giacomo,
First of all congratulations on your work!
Overall, your work is very clear, organized, and readable. I appreciate the effort you put into proposing different strategies for comparison: a simple evolutionary algorithm, one with diversity in parent selection, one with adaptability, and one with both diversity and adaptability.
Specifically, the mutation function, parent selection, and crossover functions are simple but non-trivial, and not too difficult to understand; but adding a few more comments on these functions might have improved overall readability and comprehension.
You focused extensively on comparing different strategies rather than explaining the strategies themselves and attempting to reduce fitness calls with a strategy.
The results are well presented and commented on, and the conclusions drawn are accurate and clear.
I appreciate the fact that you tested the algorithms over a high number of generations.
It would have been beneficial to have some additional analysis on how the different parameters influence the results, rather than keeping them static during the experiments.
In conclusion, I think you've done a good job. It's evident that you invested a lot of time and effort, and the result is satisfactory and aligned with the requirements of the assignment.
Good luck with the next labs!
